### PR TITLE
【Infer Symbolic Shape BUAA No.42-44】Add ops

### PR DESCRIPTION
**Pull request by Guanhuachen2003**
### PR Category
CINN

### PR Types
Others

### Description
add fake_quantize_dequantize_abs_max&&fake_quantize_dequantize_moving_average_abs_max&&fake_quantize_moving_average_abs_max ops

### CODE REVIEW

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc [hunk 1]
```diff
@@ -377,6 +377,19 @@ bool EigvalshOpInferSymbolicShape(
   return EighOpInferSymbolicShape(op, infer_context);
 }
 
+bool FakeQuantizeAbsMaxOpInferSymbolicShape(
+    pir::Operation *op, pir::InferSymbolicShapeContext *infer_context) {
+  const auto &x_shape_or_data =
+      infer_context->GetShapeOrDataForValue(op->operand_source(0));
+
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc:384]: 这里是不是可以参考infermeta添加几个attribute的约束

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc [hunk 2]
```diff
@@ -637,6 +637,49 @@ bool ConcatOpInferSymbolicShape(pir::Operation *op,
   return true;
 }
 
+bool FakeQuantizeMovingAverageAbsMaxOpInferSymbolicShape(
+    pir::Operation *op, pir::InferSymbolicShapeContext *infer_context) {
+  const symbol::ShapeOrDataDimExprs &x_shape =
+      infer_context->GetShapeOrDataForValue(op->operand_source(0));
+
+  infer_context->SetShapeOrDataForValue(op->result(0), x_shape);
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc:645]: 尽量放到函数结尾统一set

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc [hunk 3]
```diff
@@ -637,6 +637,49 @@ bool ConcatOpInferSymbolicShape(pir::Operation *op,
   return true;
 }
 
+bool FakeQuantizeMovingAverageAbsMaxOpInferSymbolicShape(
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc:640]: 解释一下这个符号推导函数在干嘛

Comment (Guanhuachen2003) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/multiary_infer_sym.cc:640]: 就是参考的原来的infermeta
void FakeQuantOrWithDequantMovingAverageAbsMaxInferMeta(
    const MetaTensor& x,
    const MetaTensor& in_scale,
    const MetaTensor& in_accum,
    const MetaTensor& in_state,
    float moving_rate,
    int bit_length,
    bool is_test,
    int round_type,
    MetaTensor* out,
    MetaTensor* out_scale,
    MetaTensor* out_state,
    MetaTensor* out_accum) {
  PADDLE_ENFORCE_EQ(bit_length >= 1 && bit_length <= 16,
                    true,
                    phi::errors::InvalidArgument(
                        "'bit_length' should be between 1 and 16, but "
                        "the received is %d",
                        bit_length));
  if (out_state) {
    out_state->set_dims({1});
  }
  if (out_accum) {
    out_accum->set_dims({1});
  }
  out->set_dims(x.dims());
  out_scale->set_dims({1});
  out->share_lod(x);
}

**Pull request closed without merging**